includes/functions/functions_customers.php

```diff
@@ -171,118 +181,144 @@ function zen_customer_greeting() {
+{
     global $db;
 
-    // -----
-    // Quick return if no customer is logged in or if the customer is in a guest checkout.
-    //
-    if (!zen_is_logged_in() || zen_in_guest_checkout()) {
-        return 0;
-    }
+    $sql = "SELECT customers_id
+            FROM " . TABLE_CUSTOMERS . "
+            WHERE customers_email_address = '" . zen_db_input($email) . "'
+            AND customers_id != " . (int)$customer_id_to_exclude;
+    $result = $db->Execute($sql);
 
-    if (is_numeric($id) == false) {
-        if (!empty($_SESSION['customer_id'])) {
-            $id = $_SESSION['customer_id'];
-        } else {
-            return 0;
-        }
+    if ($result->EOF) {
+        return true;
     }
 
-    if ($check_session == true) {
-        if (empty($_SESSION['customer_id']) || $id != $_SESSION['customer_id']) {
-            return 0;
-        }
-    }
+    return false;
+}
 
-    $addresses_query = "select count(*) as total
-                        from " . TABLE_ADDRESS_BOOK . "
-                        where customers_id = " . (int)$id;
+/**
+ * validate customer matches session
+ * @param int $customer_id
+ * @return bool
+ */
+function zen_get_customer_validate_session(int $customer_id): bool
+{
+    global $messageStack;
+    $customer = new Customer($customer_id);
 
-    $addresses = $db->Execute($addresses_query);
+    $banned = $customer->isBanned($customer_id);
 
-    return $addresses->fields['total'];
-}
+    if ($customer->isSameAsLoggedIn($customer_id) && !$banned) {
+        return true;
+    }
 
-////
-// validate customer matches session
-function zen_get_customer_validate_session($customer_id) {
-    global $db, $messageStack;
-    $zc_check_customer = $db->Execute("SELECT customers_id, customers_authorization from " . TABLE_CUSTOMERS . " WHERE customers_id=" . (int)$customer_id);
-    if ($zc_check_customer->EOF || $zc_check_customer->fields['customers_authorization'] == 4) {    // Banned status is 4
-        $db->Execute("DELETE from " . TABLE_CUSTOMERS_BASKET . " WHERE customers_id= " . $customer_id);
-        $db->Execute("DELETE from " . TABLE_CUSTOMERS_BASKET_ATTRIBUTES . " WHERE customers_id= " . $customer_id);
-        $_SESSION['cart']->reset(TRUE);
-        unset($_SESSION['customer_id']);
-        if ($zc_check_customer->fields['customers_authorization'] != 4) {
-            $messageStack->add_session('header', ERROR_CUSTOMERS_ID_INVALID, 'error');
-        }
-        return false;
+    if ($banned) {
+        $customer->resetCustomerCart();
     }
-    return true;
+
+    $messageStack->add_session('header', ERROR_CUSTOMERS_ID_INVALID, 'error');
+    return false;
 }
 
 /**
  * This function identifies whether (true) or not (false) the current customer session is
  * associated with a guest-checkout process.
+ * @alias Customer::isInGuestCheckout()
  */
-function zen_in_guest_checkout()
+function zen_in_guest_checkout(): bool
```

Comment:
Shouldn't this function be the procedural way to invoke the `Customer` class method?